api/apiclient.go

```diff
@@ -498,11 +505,14 @@ func dialWebsocket(addr, path string, opts DialOpts, tlsConfig *tls.Config, try
 	return try.Start(newWebsocketDialer(cfg, opts))
 }
 
-// newWebsocketDialer returns a function that
-// can be passed to utils/parallel.Try.Start.
-var newWebsocketDialer = createWebsocketDialer
+// newWebsocketDialer is defined as a variable so that it can be overridden
+// for tests.
+var newWebsocketDialer = newWebsocketDialer0
```

Comment:
This is something that already exists. I would prefer not to have to fix everything just because it's been touched.

Also, as expressed in our chat online, this is very much an implementation detail - the tests that use this
are checking that some particular logic has been wired up correctly. It wouldn't make sense for external
code to swap out this particular implementation detail.

If we *were* to make this an external dependency, I'd be more inclined to make the dependency
a higher level thing (something that returns an rpc.Codec for example), but that's something that
would better suit a separate PR, I think.